test/pytest/test_token_authorization.py

```diff
@@ -0,0 +1,212 @@
+
+
+# Test inference API with token enabled
+def test_inference_api_with_token():
+    key = read_key_file("inference")
+    header = {"Authorization": f"Bearer {key}"}
+
+    response = requests.post(
+        url="http://localhost:8080/predictions/resnet18",
+        files={"data": open(data_file_kitten, "rb")},
+        headers=header,
+    )
+    time.sleep(5)
+    print(response.text)
+    print(key)
+
+    assert response.status_code == 200, "Token check failed"
+
+
+# Test inference API with incorrect token
+def test_inference_api_with_incorrect_token():
+    # Using random key
+    header = {"Authorization": "Bearer abcd1234"}
+
+    response = requests.post(
+        url="http://localhost:8080/predictions/resnet18",
+        files={"data": open(data_file_kitten, "rb")},
+        headers=header,
+    )
+    time.sleep(5)
+    print(response.text)
+
+    assert response.status_code == 400, "Token check failed"
+
+
+# Test Token API for regenerating new inference key
+def test_token_inference_api():
+    token_key = read_key_file("token")
+    inference_key = read_key_file("inference")
+    header_inference = {"Authorization": f"Bearer {inference_key}"}
+    header_token = {"Authorization": f"Bearer {token_key}"}
+    params = {"type": "inference"}
+
+    # check inference works with current token
+    response = requests.post(
+        url="http://localhost:8080/predictions/resnet18",
+        files={"data": open(data_file_kitten, "rb")},
+        headers=header_inference,
+    )
+    time.sleep(5)
+    assert response.status_code == 200, "Token check failed"
+
+    # generate new inference token and check it is different
+    response = requests.get(
+        url="http://localhost:8081/token", params=params, headers=header_token
+    )
+    time.sleep(5)
+    print(response.text)
+    print(token_key)
+    assert response.status_code == 200, "Token check failed"
+    assert inference_key != read_key_file("inference"), "Key file not updated"
+
+    # check inference does not works with original token
+    response = requests.post(
+        url="http://localhost:8080/predictions/resnet18",
+        files={"data": open(data_file_kitten, "rb")},
+        headers=header_inference,
+    )
+    time.sleep(5)
+    assert response.status_code == 400, "Token check failed"
+
+
+# Test Token API for regenerating new management key
+def test_token_management_api():
+    token_key = read_key_file("token")
+    management_key = read_key_file("management")
+    header = {"Authorization": f"Bearer {token_key}"}
+    params = {"type": "management"}
+
+    response = requests.get(
+        url="http://localhost:8081/token", params=params, headers=header
+    )
+    time.sleep(5)
+    print(response.text)
+    print(token_key)
+
+    assert management_key != read_key_file("management"), "Key file not updated"
+    assert response.status_code == 200, "Token check failed"
+    test_utils.stop_torchserve()
```

Comment:
See comment about module scope fixture for this.

Reply:
Would be good to actually test token expiration as well. Just setup expiration to a low value and run two requests while waiting in-between for token to expire.